extensions\apidoc\helper\ApiIndexer.php - add Unicode support FileInfo(jsSearch) for bootstrap-guide mode.

keltstr opened the pull request:
extensions\apidoc\helper\ApiIndexer.php - add Unicode support FileInfo(jsSearch) for bootstrap-guide mode.

```
preg_match(/(.*)/s)
```

replaced 
    preg_match(/(._|\r_|\n*)/u)

keltstr committed:
extensions\apidoc\helper\ApiIndexer.php - add Unicode support FileInfo(jsSearch) for bootstrap-guide mode.
preg_match(/(.*)/s) replaced preg_match(/(.*|\r*|\n*)/u)

samdark commented on extensions/apidoc/helpers/ApiIndexer.php:
```diff
@@ -25,18 +25,18 @@ class ApiIndexer extends Indexer
-        // create file entry
-        if (preg_match('~<h1>(.*?)</h1>~s', $contents, $matches)) {
+		// create file entry
+        if (preg_match('~<h1>(.*|\r*|\n*?)</h1>~u', $contents, $matches)) {
```
Shouldn't `.*` include `\r` and `\n`?

samdark replied:
If not, shouldn't `\R` be used instead?

cebe replied:
this has been handled by `/s` modifier before: http://php.net/manual/de/reference.pcre.pattern.modifiers.php
also adding `|` always needs new parenthesis: `(.*|\r*|\n*)`

cebe commented:
What are you trying to solve with this change? .* should match any unicdeo character even without `u`...

keltstr commented:
preg_match don't match utf8 without /u modifier (I was generating russian(utf8) guide with apidoc - and having empty fileInfo in jssearch index).
But /u and /s modifiers don't work together (/su or /us) - so I added /s modifier functionality using .|\r|\n and added /u modifier (tested ok)

keltstr commented:
In the process of Guide generating is created file jssearch.index.js - it should contain `jssearch.files` variable, but after generating of Russian Guide(utf8) `jssearch.files` remains empty (jssearch.files="";). In English Guide `jssearch.files` normaly contains info about files and it used for output listing search result on a page.

cebe commented:
I have reproduced the issue, if you enable PHP warnings in your php config which you definitively should, you get an exception about invalid UTF8.
